app/src/models/status.ts

```diff
@@ -33,6 +33,31 @@ export enum AppFileStatus {
   Resolved = 'Resolved',
 }
 
+/** The state of a GitHub Desktop-specific change containing additional metadata */
+export type PlainFileStatus = {
+  kind:
+    | AppFileStatusKind.New
+    | AppFileStatusKind.Modified
+    | AppFileStatusKind.Deleted
+    // TODO: where should this live?
+    | AppFileStatusKind.Resolved
```

Comment:
what are you thoughts here, @shiftkey 
?

i have mixed feelings about putting this under `PlainFileStatus`. these are all files that can exist in a conflicted merge but require no action. but, they all can appear in a regular status, except for resolved.

what if we have two overlapping type unions? one for regular statuses and one for files in conflict that require no action?

```ts
| AppFileStatusKind.New
| AppFileStatusKind.Modified
| AppFileStatusKind.Deleted
```

and 

```ts
| AppFileStatusKind.New
| AppFileStatusKind.Modified
| AppFileStatusKind.Deleted
| AppFileStatusKind.Resolved
```

(or something to that effect)

Reply:
I left it there as a comment to unblock this PR, because I knew I'd need to think about this later.

Rather than go too deep here I'd suggest we figure it out in #6214 what feels more natural.

Reply:
if this is going into `master` and its got a `todo` comment on it and we're not sure where it should live, i'd prefer to have it be its own type in the interim. non-blocking.

Reply:
@outofambit let me see what I can do here without introducing too much unnecessary churn

Reply:
bedd01ace